src/Microsoft.Performance.SDK.Runtime/ExtensionRoot.cs

```diff
@@ -0,0 +1,189 @@
+
+        /// <inheritdoc />
+        public IEnumerable<DataProcessorId> DataProcessors => this.Repository.DataProcessors;
+
+        /// <inheritdoc />
+        public bool AddCompositeDataCookerReference(ICompositeDataCookerReference dataCooker)
+        {
+            return this.Repository.AddCompositeDataCookerReference(dataCooker);
+        }
+
+        /// <inheritdoc />
+        public bool AddDataProcessorReference(IDataProcessorReference dataProcessorReference)
+        {
+            return this.Repository.AddDataProcessorReference(dataProcessorReference);
+        }
+
+        /// <inheritdoc />
+        public bool AddSourceDataCookerReference(ISourceDataCookerReference dataCooker)
+        {
+            return this.Repository.AddSourceDataCookerReference(dataCooker);
+        }
+
+        /// <inheritdoc />
+        public bool AddTableExtensionReference(ITableExtensionReference tableExtensionReference)
+        {
+            return this.Repository.AddTableExtensionReference(tableExtensionReference);
+        }
+
+        /// <inheritdoc />
+        public void Dispose()
+        {
+            this.Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <inheritdoc />
+        public void FinalizeDataExtensions()
+        {
+            this.Repository.FinalizeDataExtensions();
+            this.dependencyGraph = DependencyDag.Create(this.Catalog, this.Repository);
+        }
+
+        /// <inheritdoc />
+        public ICompositeDataCookerReference GetCompositeDataCookerReference(DataCookerPath dataCookerPath)
+        {
+            return this.Repository.GetCompositeDataCookerReference(dataCookerPath);
+        }
+
+        /// <inheritdoc />
+        public IDataProcessorReference GetDataProcessorReference(DataProcessorId dataProcessorId)
+        {
+            return this.Repository.GetDataProcessorReference(dataProcessorId);
+        }
+
+        /// <inheritdoc />
+        public ISourceDataCookerFactory GetSourceDataCookerFactory(DataCookerPath dataCookerPath)
+        {
+            return this.Repository.GetSourceDataCookerFactory(dataCookerPath);
+        }
+
+        /// <inheritdoc />
+        public ISourceDataCookerReference GetSourceDataCookerReference(DataCookerPath dataCookerPath)
+        {
+            return this.Repository.GetSourceDataCookerReference(dataCookerPath);
+        }
+
+        private void Dispose(bool disposing)
+        {
+            if (this.isDisposed)
+            {
+                return;
+            }
+
+            if (disposing)
+            {
+                try
+                {
+                    var dag = DependencyDag.Create(this.Catalog, this.Repository);
+                    dag.DependentWalk(
+                        x => x.Target.Match(
+                            r => r.SafeDispose(),
+                            r => r.SafeDispose()));
+                }
```

Comment:
It seems like we should catch/eat any exception here - given that this is in a Dispose method

Reply:
I'll update it.